### ISSUE TITLE
Improve inbox by taking per-user data into account

### ISSUE THREAD

**Issue by GoogleCodeExporter** [strikersix23/walkaround#43]
```
The indexing logic needs to take per-user data such as read/unread state and 
archived state into account so that /inbox can e.g. highlight unread waves or 
hide archived waves.
```

Original issue reported on code.google.com by `oh...@google.com` on 13 Jan 2012 at 7:27

**Comment by GoogleCodeExporter** [comment]
Original comment by `oh...@google.com` on 16 Jan 2012 at 8:36
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `oh...@google.com` on 19 Jan 2012 at 8:22

**Comment by GoogleCodeExporter** [comment]
```
Dan fixed this in 08ddcb60e899.  (Archiving is not yet implemented, but the 
general problem that this bug was about is fixed.)
```

Original comment by `oh...@google.com` on 2 Feb 2012 at 10:38
- Changed state: **Fixed**
